include/openPMD/backend/Attribute.hpp

```diff
@@ -199,7 +200,6 @@ auto doConvert(T *pv) -> U
             "getCast: no scalar to vector conversion possible.");
     }
 
-    (void)pv;
     throw std::runtime_error("getCast: no cast possible.");
```

Comment:
Need to put these into an else branch, ICC seems to determine unreachable statements after resolving if constexpr

Reply:
ah classic. a range of NVCC 11 minor releases does that as well